internal/pipe/checksums/checksums.go

```diff
@@ -121,6 +125,34 @@ func refresh(ctx *context.Context, filepath string) error {
 		return err
 	}
 
+	artifactsMapping := make(map[string]int)
+	for i, a := range artifactList {
+		if a.Type == artifact.UploadableFile {
+			continue // skip extra files
+		}
+		// Path is the only unique field for an artifact
+		artifactsMapping[a.Path] = i
+	}
+
+	err = ctx.Artifacts.Visit(func(a *artifact.Artifact) error {
+		if index, ok := artifactsMapping[a.Path]; ok {
+			sumFields := strings.Fields(sumLines[index])
+			if len(sumFields) == 0 {
+				return fmt.Errorf("missing checksum in sumline of %s: %s", a.Path, sumFields)
+			}
+			sum := sumFields[0]
+			if a.Extra == nil {
+				a.Extra = make(artifact.Extras)
+			}
+			a.Extra[artifactChecksumExtra] = fmt.Sprintf("%s:%s", ctx.Config.Checksum.Algorithm, sum)
```

Comment:
can we put this in the `checksum` func instead? I think we then don't need most of the rest of the code and it should simplify things quite a bit...

Reply:
That makes total sense. Not sure why I took the visit in the first place.
done (also applied fixes from the lint)